### PR TITLE
Add support for python3.7

### LINKED ISSUE #87: Python 3.7 is not supported
Eventhough the pyproject.toml has `requires-python = ">=3.7"`, the type hinting used in some of the files does not work with Python 3.7 where `typing_extensions` is needed to import `Literal`.

https://github.com/astronomer/astronomer-cosmos/blob/4edb446afbbd8166362d9fe29e421637821aba0d/cosmos/providers/dbt/dag.py#L4

https://github.com/astronomer/astronomer-cosmos/blob/4edb446afbbd8166362d9fe29e421637821aba0d/cosmos/providers/dbt/render.py#L5

### PR DESCRIPTION
**Pull request by jbandoro**
Fixes #87

Type hinting updates so can support Python 3.7

Ran the pre-commit hook which did isort and black updates, but there were a lot of E501 Line too long errors that I ignored.

### CODE REVIEW

File: cosmos/providers/dbt/dag.py
Comment (jlaneve) [inline cosmos/providers/dbt/dag.py:7]: do we need to add `typing_extensions` to the requirements?

**Review by jlaneve — CHANGES_REQUESTED** [review]
Thank you for the PR @jbandoro! Very much appreciated.

Just left a small comment around adding `typing_extensions` to the pyproject.toml requirements. We use [Hatch](https://hatch.pypa.io/latest/) as a build system, and there's [docs here](https://hatch.pypa.io/latest/config/dependency/#environment-markers) around how to use environment markers to only install if the Python version <= 3.7.

File: cosmos/providers/dbt/dag.py
Comment (jbandoro) [reply cosmos/providers/dbt/dag.py:7]: `typing_extensions` is [installed](https://github.com/apache/airflow/blob/main/setup.cfg#L137) with `apache-airflow` which is already a dependency in the pyproject.toml. I'd be happy to add it here too if you think that's a good idea.

Comment (jlaneve) [reply cosmos/providers/dbt/dag.py:7]: Hmm, let's add it just to be explicit & safe if you don't mind

**Review by jlaneve — APPROVED** [review]
LGTM! When you merge the PR, I'll release a new version.

Separately - let us know how you like Cosmos! If there's anything missing for your use case, file an issue and we can help work on it. And if you want help setting it up or just want to chat about our plans for Cosmos, feel free to drop me a note at julian@astronomer.io 🙂

**Comment by jbandoro** [comment]
> LGTM! When you merge the PR, I'll release a new version.

> Separately - let us know how you like Cosmos! If there's anything missing for your use case, file an issue and we can help work on it. And if you want help setting it up or just want to chat about our plans for Cosmos, feel free to drop me a note at [julian@astronomer.io](mailto:julian@astronomer.io) 🙂

Yeah I saw Cosmos posted on LinkedIn and the ability to parse a DAG from a dbt project is super cool! I wanted to try it out in my team's test environment but we're still using Python 3.7. 

I'm going to hold off on merging, since I'm still trying to get a `DbtTaskGroup` to run from my feature branch, so once I confirm I can get a dbt model parsed and running I'll confirm here or make any needed updates for Python3.7. Thanks!

**Comment by jbandoro** [comment]
> I'm going to hold off on merging, since I'm still trying to get a DbtTaskGroup to run from my feature branch, so once I confirm I can get a dbt model parsed and running I'll confirm here or make any needed updates for Python3.7. Thanks!

Got a dbt dag to run with python 3.7 🥳, the commit above was the only issue for unioning 2 dictionaries in python < 3.9 - thanks for your team's work on cosmos it's really cool!

**Comment by jbandoro** [comment]
Also I can't merge in this repo, so please go ahead if the update looks good.

**Pull request merged**
